arcaflow_plugin_fio/fio_schema.py

```diff
@@ -56,110 +65,431 @@ def is_sync(self) -> bool:
-        }
-    )
+    # Units
+    kb_base: typing.Annotated[
+        typing.Optional[KbBase],
+        schema.name("Units Base"),
+        schema.description(
+            "Select the interpretation of unit prefixes in input parameters. The "
+            "default is 1024 (compatibility mode): Power-of-2 values with SI prefixes "
+            "(e.g., KB) and Power-of-10 values with IEC prefixes (e.g., KiB). When "
+            "setting this to 1000, inputs comply with IEC 80000-13 and the "
+            "International System of Units (SI): Power-of-2 values with IEC prefixes "
+            "(e.g., KiB) and Power-of-10 values with SI prefixes (e.g., KB)"
+        ),
+    ] = None
+
+    # Job Description
+    loops: typing.Annotated[
+        typing.Optional[int],
+        schema.name("Number of Job Loops"),
+        schema.description(
+            "Run the specified number of iterations of this job. Used to repeat the "
+            "same workload a given number of times. Default is 1."
+        ),
+    ] = None
+    numjobs: typing.Annotated[
+        typing.Optional[int],
+        schema.name("Number of Job Clones"),
+        schema.description(
+            "Create the specified number of clones of this job. Each clone of job is "
+            "spawned as an independent thread or process. May be used to set up a "
+            "larger number of threads/processes doing the same thing. Each thread is "
+            "reported separately. Default is 1."
+        ),
+    ] = None
+
+    # Time related parameters
+    runtime: typing.Annotated[
+        typing.Optional[str],
+        validation.pattern(time_pattern),
+        schema.name("Job Run Time"),
+        schema.description(
+            "Limit runtime. The test will run until it completes the configured I/O "
+            "workload or until it has run for this specified amount of time, whichever "
+            "occurs first. When the unit is omitted, the value is interpreted in "
+            "seconds. Default behavior is size-based operation."
+        ),
+    ] = None
+    time_based: typing.Annotated[
+        typing.Optional[bool],
+        schema.name("Time Based"),
+        schema.description(
+            "If set, fio will run for the duration of the runtime specified even if "
+            "the file(s) are completely read or written. It will simply loop over the "
+            "same workload as many times as the runtime allows. Default is false."
+        ),
+    ] = None
+    startdelay: typing.Annotated[
+        typing.Optional[str],
+        schema.name("Job Start Delay"),
+        schema.description(
+            "Delay the start of job for the specified amount of time. Can be a single "
+            "value or a range. When given as a range, each thread will choose a value "
+            "randomly from within the range. Value is in seconds if a unit is omitted. "
+            "Default is 0."
+        ),
+    ] = None
```

Comment:
Should this input be validated against `time_pattern`?

Reply:
Actually, this one is even more complicated because it can be a range of durations. :-/